Source/WebCore/html/HTMLElement.cpp

```diff
@@ -834,8 +834,7 @@ std::optional<TextDirection> HTMLElement::directionalityIfDirIsAuto() const
 auto HTMLElement::computeDirectionalityFromText() const -> TextDirectionWithStrongDirectionalityNode
 {
     if (auto* textControl = dynamicDowncast<HTMLTextFormControlElement>(const_cast<HTMLElement*>(this))) {
-        auto* inputElement = dynamicDowncast<HTMLInputElement>(textControl);
-        if (!inputElement || (inputElement->isTextType() && !inputElement->isPasswordField())) {
+        if (textControl->dirAutoUsesValue()) {
```

Comment:
This seems like a behavior change to include `<textarea>` – do we need to add a test for that?

Reply:
It's not a behavior change, `!inputElement` covers `<textarea>`